File: module6/crud/fe/product/src/app/model/product.ts
import {Category} from './category';

export interface Product {
  id?: number;
  name?: string;
  made?: string;
  dayImport?: string;
  expire?: string;
  number?: number;
  price?: number;
  image?: string;
  isDeleted?: boolean;
  category?: Category;
}

File: module6/crud/be/product/src/main/java/com/example/model/Product.java
package com.example.model;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String made;
    private String dayImport;
    private String expire;
    private Integer number;
    private Double price;
    private String image;
    @Column(columnDefinition = "bit(1) default 0")
    private Boolean isDeleted;

    @ManyToOne()
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;
}
